zebra-test/src/net.rs

```diff
@@ -47,37 +48,39 @@ pub fn zebra_skip_ipv6_tests() -> bool {
     false
 }
 
-/// Returns a random port number from the ephemeral port range.
+/// Uses the "magic" port number that tells the operating system to
+/// choose a random unallocated port.
+///
+/// The OS chooses a different port each time it opens a connection or
+/// listener with this magic port number.
 ///
-/// Does not check if the port is already in use. It's impossible to do this
-/// check in a reliable, cross-platform way.
+/// Creates a TcpListener to find a random unallocated port, then drops the TcpListener to close the socket.
+///
+/// Returns the unallocated port number.
 ///
 /// ## Usage
 ///
 /// If you want a once-off random unallocated port, use
 /// `random_unallocated_port`. Don't use this function if you don't need
-/// to - it has a small risk of port conflcits.
+/// to - it has a small risk of port conflicts when there is a delay
+/// between this fn call and binding the tcp listener.
 ///
 /// Use this function when you need to use the same random port multiple
 /// times. For example: setting up both ends of a connection, or re-using
 /// the same port multiple times.
+///
+/// ## Panics
+///
+/// If the OS finds no available ports
+///
+/// If there is an OS error when getting the socket address
 pub fn random_known_port() -> u16 {
-    // Use the intersection of the IANA/Windows/macOS ephemeral port range,
-    // and the Linux ephemeral port range:
-    //   - https://en.wikipedia.org/wiki/Ephemeral_port#Range
-    // excluding ports less than 53500, to avoid:
-    //   - typical Hyper-V reservations up to 52000:
-    //      - https://github.com/googlevr/gvr-unity-sdk/issues/1002
-    //      - https://github.com/docker/for-win/issues/3171
-    //   - the MOM-Clear port 51515
-    //      - https://docs.microsoft.com/en-us/troubleshoot/windows-server/networking/service-overview-and-network-port-requirements
-    //   - the LDAP Kerberos byte-swapped reservation 53249
-    //      - https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/ldap-kerberos-server-reset-tcp-sessions
-    //   - macOS and Windows sequential ephemeral port allocations,
-    //     starting from 49152:
-    //      - https://dataplane.org/ephemeralports.html
-
-    rand::thread_rng().gen_range(53500..60999)
+    let host_ip = Ipv4Addr::new(127, 0, 0, 1);
```

Comment:
If we get port conflicts in tests, we might want to try 0.0.0.0 here, because some tests want to bind to all addresses.

But let's see how it goes first? Maybe 127.0.0.1 is better.